src/builtins/string.cpp

```diff
@@ -1,1946 +0,0 @@
-        streams.err.append_format(BUILTIN_ERR_COMBO2, cmd,
-                                  _(L"--invert and --groups-only are mutually exclusive"));
-        return STATUS_INVALID_ARGS;
-    }
-
-    if (opts.entire && opts.groups_only) {
-        streams.err.append_format(BUILTIN_ERR_COMBO2, cmd,
-                                  _(L"--entire and --groups-only are mutually exclusive"));
-        return STATUS_INVALID_ARGS;
-    }
-
-    std::unique_ptr<string_matcher_t> matcher;
-    if (!opts.regex) {
-        // Globs cannot fail.
-        matcher = make_unique<wildcard_matcher_t>(pattern, opts);
-    } else {
-        // Compile the pattern as regex and validate capture group names as variables; both may
-        // fail. Note both try_compile_regex and validate_capture_group_names print an error on
-        // failure.
-        auto re = try_compile_regex(pattern, opts, cmd, streams);
-        if (!re || !validate_capture_group_names(re->capture_group_names(), streams)) {
-            return STATUS_INVALID_ARGS;
-        }
-        matcher = make_unique<regex_matcher_t>(re.acquire(), opts);
-    }
-
-    assert(matcher && "Should have a matcher");
-    arg_iterator_t aiter(argv, optind, streams);
-    while (const wcstring *arg = aiter.nextstr()) {
-        matcher->report_matches(*arg, streams);
-        if (opts.quiet && matcher->match_count() > 0) {
-            break;
-        }
-    }
-    matcher->import_captures(parser.vars());
-
-    return matcher->match_count() > 0 ? STATUS_CMD_OK : STATUS_CMD_ERROR;
-}
-
-static int string_pad(parser_t &parser, io_streams_t &streams, int argc, const wchar_t **argv) {
-    options_t opts;
-    opts.char_to_pad_valid = true;
-    opts.right_valid = true;
-    opts.width_valid = true;
-    int optind;
-    int retval = parse_opts(&opts, &optind, 0, argc, argv, parser, streams);
-    if (retval != STATUS_CMD_OK) return retval;
-
-    size_t pad_char_width = fish_wcwidth(opts.char_to_pad);
-    if (pad_char_width == 0) {
-        string_error(streams, _(L"%ls: Invalid padding character of width zero\n"), argv[0]);
-        return STATUS_INVALID_ARGS;
-    }
-
-    // Pad left by default
-    if (!opts.right) {
-        opts.left = true;
-    }
-
-    // Find max width of strings and keep the inputs
-    ssize_t max_width = 0;
-    std::vector<wcstring> inputs;
-
-    arg_iterator_t aiter_width(argv, optind, streams);
-    while (const wcstring *arg = aiter_width.nextstr()) {
-        wcstring input_string = *arg;
-        ssize_t width = width_without_escapes(input_string);
-        if (width > max_width) max_width = width;
-        inputs.push_back(std::move(input_string));
-    }
-
-    ssize_t pad_width = max_width > opts.width ? max_width : opts.width;
-    for (auto &input : inputs) {
-        wcstring padded;
-        ssize_t padded_width = width_without_escapes(input);
-        if (pad_width >= padded_width) {
-            ssize_t pad = (pad_width - padded_width) / pad_char_width;
```

Comment:
ok you fixed the weird variable name of `padded_width` and the redundant if statement